packages/rich-text/src/register-format-type.js

```diff
@@ -124,5 +113,24 @@ export function registerFormatType( name, settings ) {
 
 	dispatch( 'core/rich-text' ).addFormatTypes( settings );
 
+	if (
+		settings.__experimentalCreatePrepareEditableTree &&
+		settings.__experimentalGetPropsForEditableTreePreparation
+	) {
+		addFilter( 'experimentalRichText', name, ( OriginalComponent ) => {
+			return withSelect( ( sel ) => ( {
+				[ `format_${ name }` ]: settings.__experimentalGetPropsForEditableTreePreparation( sel ),
+			} ) )( ( props ) => (
+				<OriginalComponent
+					{ ...props }
+					prepareEditableTree={ [
```

Comment:
We are generating a new array here on each render, I wonder if this is the source of the rerendering happening in all RichText. I wonder if we could memoize it somehow in `withSelect` by using `settings.__experimentalGetPropsForEditableTreePreparation( sel )`  as the memoization key, as we don't want to generate a new array if the the returned value didn't change.

Anyway, it doesn't feel unsolvable, so for the sake of iterations, I'm fine leaving this optimisation for a follow-up.

Reply:
Could it also be creating a new object in `withSelect` every time? But sure, sounds resolvable.

Reply:
Should we create an issue for this one?